### PR TITLE
Win32: Spawn extension processes in job

### PR DESCRIPTION
**Pull request by mook-as**
This puts any processes spawned by the extension in a job, so that they can be terminated when the main application exits.

We intentionally let the main Rancher Desktop process have a handle to the new job (which isn't ever referenced), and set the job up so that it will automatically terminate when all handles to that job are closed.  This means that as the main process exits, it will automatically cause all processes in that job to terminate (because it would have the only outstanding handle to the job).

For #7653

### CODE REVIEW

File: src/go/rdctl/pkg/process/process_windows.go [hunk 1]
```diff
@@ -28,14 +29,244 @@ import (
...
+			JobObjectExtendedLimitInformation,
+			uintptr(unsafe.Pointer(&limits)),
+			unsafe.Sizeof(limits))
+		if ok == 0 {
+			return nil, fmt.Errorf("error setting job limits: %w", err)
+		}
+
+		// Duplicate the job into the given process (but leaking it).  This way when
+		// the target process exits, it will shut down the job.
+		hProc, err := windows.OpenProcess(windows.PROCESS_DUP_HANDLE, false, pid)
```
Comment (Nino-K) [inline src/go/rdctl/pkg/process/process_windows.go:240]: Also, 
```
// injectJobHandleIntoParent duplicates the job handle into the given parent process
// so the parent can manage the job's lifecycle.
func injectJobHandleIntoParent(pid uint32, job windows.Handle) error {
    hProc, err := windows.OpenProcess(windows.PROCESS_DUP_HANDLE, false, pid)
    if err != nil {
        return fmt.Errorf("failed to open parent process %d: %w", pid, err)
    }
    defer windows.CloseHandle(hProc)

    err = windows.DuplicateHandle(windows.CurrentProcess(), job, hProc, nil, 0, false, 0)
    if err != nil {
        return fmt.Errorf("failed to duplicate job handle: %w", err)
    }

    return nil
}
```

File: src/go/rdctl/pkg/process/process_windows_test.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/windows"
+)
+
+func TestBuildCommandLine(t *testing.T) {
+	t.Parallel()
+	cases := [][]string{
+		{"arg0", "a b c", "d", "e"},
+		{"C:\\Program Files\\arg0\\\\", "ab\"c", "\\", "d"},
```
Comment (Nino-K) [inline src/go/rdctl/pkg/process/process_windows_test.go:16]: Should there also be a test for mixed forward and backward slashes or is that too crazy?
e.g `C:/Program Files\\Test`

File: src/go/rdctl/pkg/process/process_windows_test.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/windows"
+)
+
+func TestBuildCommandLine(t *testing.T) {
+	t.Parallel()
+	cases := [][]string{
```
Comment (Nino-K) [inline src/go/rdctl/pkg/process/process_windows_test.go:14]: There are some more cases that I can think of, but I will leave it up to you to see if they make sense or not:
- Spaces in the middle of the argument
- Trailing spaces (extra spaces after quotes, etc)
- Windows-specific characters that are valid in the command line and have special meaning e.g. `&, |, >, <`, etc.

File: src/go/rdctl/pkg/process/process_windows.go [hunk 4]
```diff
@@ -28,14 +29,244 @@ import (
...
+
 // TerminateProcessInDirectory terminates all processes where the executable
 // resides within the given directory, as gracefully as possible.  If `force` is
 // set, SIGKILL is used instead.
 func TerminateProcessInDirectory(directory string, force bool) error {
 	var pids []uint32
 	// Try EnumProcesses until the number of pids returned is less than the
 	// buffer size.
-	err := directories.InvokeWin32WithBuffer(func(size int) error {
+	err := directories.InvokeWin32WithBuffer(256, func(size int) error {
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/go/rdctl/pkg/process/process_windows.go]: Turns out we didn't need a custom initial size, so I got rid of it again.

File: src/go/rdctl/pkg/process/process_windows_test.go (hunk 2 above)
Comment (mook-as) [reply src/go/rdctl/pkg/process/process_windows_test.go:16]: Added test. This also happened to catch an error when there are no arguments :)

**Review by Nino-K — APPROVED** [review]

**Pull request merged**